lnwallet/channel.go

```diff
@@ -5532,6 +5532,39 @@ func (lc *LightningChannel) NumLocalUpdatesPendingOnRemote() uint64 {
 	return lc.localUpdateLog.logIndex - lastRemoteCommit.ourMessageIndex
 }
 
+// NumLocalUpdatesPendingOnLocal returns the number of local updates that still
+// need to be applied to the local commitment tx.
+func (lc *LightningChannel) NumLocalUpdatesPendingOnLocal() uint64 {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	lastLocalCommit := lc.localCommitChain.tip()
+
+	return lc.localUpdateLog.logIndex - lastLocalCommit.ourMessageIndex
```

Comment:
If IIUC, this method is returning the num of updates which, we've sent, yet we are waiting for an ack from the remote node. In that case I think it's more accurate to use `remoteCommitChain.tail.ourMessageIndex`?

Reply:
No we have to call this to figure out if we are eligible to send `stfu`. All local updates have to be both signed for (signed on remote commitment) and counter-signed (signed on local commitment) before we can send `stfu` per the protocol rules.